front/src/common/IntervalsDataViz/IntervalItem.tsx

```diff
@@ -56,21 +56,25 @@ const IntervalItem = <T extends { [key: string]: string | number }>({
       valueText = `${interval[field]}`;
     }
   }
+
+  const isNoData =
+    !!field &&
+    (segment === undefined || segment[field] === undefined || segment[field] === emptyValue);
+  const isHasData =
+    !!field &&
+    segment !== undefined &&
+    segment[field] !== undefined &&
+    segment[field] !== emptyValue;
+  const fieldValue = !!field && segment !== undefined && segment[field];
+  const isDataZero = fieldValue === 0;
+
   return (
     <div
       className={cx(
         'item',
         highlighted.includes(segment.index) && 'highlighted',
-        field &&
-          segment !== undefined &&
-          segment[field] !== undefined &&
-          segment[field] !== emptyValue &&
-          'with-data',
-        field &&
-          (segment === undefined ||
-            segment[field] === undefined ||
-            segment[field] === emptyValue) &&
-          'no-data',
+        isHasData && 'with-data',
+        isNoData && 'no-data',
```

Comment:
We have a convention when using condition in cx
```suggestion
        { 
          'with-data': isHasData,
          'no-data': isNoData,
        },
```